src/zope/app/testing/xmlrpc.rst

```diff
@@ -0,0 +1,392 @@
+===============
+ XML-RPC views
+===============
+
+XML-RPC Methods
+===============
+
+There are two ways to write XML-RPC views. You can write views that
+provide "methods" for other objects, and you can write views that have
+their own methods.  Let's look at the former case first, since it's a
+little bit simpler.
+
+Let's write a view that returns a folder listing:
+
+  >>> class FolderListing(object):
+  ...     def contents(self):
+  ...         return list(self.context.keys())
```

Comment:
Perhaps `sorted(self.context.keys())`?

Reply:
So `xmlrpc.rst` was copied directly from https://github.com/zopefoundation/zope.app.publisher/blob/master/src/zope/app/publisher/xmlrpc/README.rst 

Perhaps I should leave a note about that.

All I changed was removing some initial belt-and-suspenders test code I used when debugging the z.a.publisher tests.

It's funny: `xmlrpc.py` wound up getting copied into z.a.publisher when I was breaking the dependency between the two and now I nearly copied it back when adding Python 3 support.